packages/xpetra/src/CMakeLists.txt

```diff
@@ -16,9 +16,88 @@ TRIBITS_CONFIGURE_FILE(${PACKAGE_NAME}_config.hpp)
 # src
 #
 
+############################################################################################
+FUNCTION(XPETRA_PROCESS_ETI_TEMPLATE ETI_CLASSES TEMPLATE_FILE PROCESSED_FILE SOURCES_LIST)
+  SET(SRCS "")
+  FOREACH(CLASS ${ETI_CLASSES})
+    # find spearator of class name and (optional) conditions
+    string(FIND ${CLASS} "-" DASHPOS)
+    IF ('${DASHPOS}' STREQUAL '-1')
+      string(SUBSTRING ${CLASS} 0 ${DASHPOS} CLASS_NAME)
+      #MESSAGE("New class name = ${CLASS_NAME}")
+      set (CONDITION_NAME "")
+      set (CONDITION_NAME_END "")      
+    ELSE()
+      string(SUBSTRING ${CLASS} 0 ${DASHPOS} CLASS_NAME)
+      #MESSAGE("New class name = ${CLASS_NAME}")
+      string(SUBSTRING ${CLASS} ${DASHPOS} -1 CONDITION_NAME)
+      string(REPLACE "[" "(" CONDITION_NAME ${CONDITION_NAME})
+      string(REPLACE "]" ")" CONDITION_NAME ${CONDITION_NAME})
+      string(REPLACE "." " " CONDITION_NAME ${CONDITION_NAME})
+      string(REPLACE "-" "" CONDITION_NAME ${CONDITION_NAME})
+      string(REPLACE "?" "#" CONDITION_NAME ${CONDITION_NAME})
+      string(STRIP CONDITION_NAME ${CONDITION_NAME})
+      #MESSAGE("New condition name = ${CONDITION_NAME}")
+      set (CONDITION_NAME_END "#endif")
+    ENDIF()
+    
+    string(REPLACE "::" "_" CLASS_FILE_NAME "${CLASS_NAME}")
```

Comment:
I wonder if we should move the "mangle class name for use in macro" function (rewritten in a few places, perhaps with differences) up into Teuchos, where different packages could use it.